### PR TITLE
Add more statement parsing for `sass-parser`

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: pkg/sass-parser/lib/src/stringifier.ts [hunk 1]
```diff
@@ -67,7 +70,67 @@ export class Stringifier extends PostCssStringifier {
     )(statement, semicolon);
   }
 
+  private ['debug-rule'](node: DebugRule, semicolon: boolean): void {
+    this.builder(
+      '@debug' +
+        (node.raws.afterName ?? ' ') +
```
Comment (Goodwine) [inline pkg/sass-parser/lib/src/stringifier.ts:76]: I just realized that setting `afterName = ''` in the case of `@debug foo` would result in `@debugfoo`. I was worried about that, but then I tried it on postcss and that's the same behavior! o_O

Interestingly, in postcss, if I set `aftername = null`, this results in `@debugnullfoo` 🤔

File: pkg/sass-parser/lib/src/stringifier.ts [hunk 2]
```diff
@@ -67,7 +70,67 @@ export class Stringifier extends PostCssStringifier {
...
+      node,
+      '@for' +
+        (node.raws.afterName ?? ' ') +
+        node.params +
+        (node.raws.between ?? '')
+    );
+  }
+
   private atrule(node: GenericAtRule, semicolon: boolean): void {
+    // If the @at-root shorthand, stringify `@at-root {.foo {...}}` as
```
Comment (Goodwine) [inline pkg/sass-parser/lib/src/stringifier.ts:116]: It is a bit awkward to read, but I'm not sure how to word it differently 🤔

**Review by Goodwine — CHANGES_REQUESTED** [review]

File: pkg/sass-parser/lib/src/statement/index.ts [hunk 3]
```diff
@@ -30,14 +37,29 @@ export type AnyStatement = Root | Rule | GenericAtRule;
...
 /**
  * All Sass statements that are also at-rules.
  *
  * @category Statement
  */
-export type AtRule = GenericAtRule;
+export type AtRule = DebugRule | EachRule | ErrorRule | ForRule | GenericAtRule;
+
+/**
+ * All Sass statements that are comments.
```
Comment (Goodwine) [inline pkg/sass-parser/lib/src/statement/index.ts:58]: Is this plural because we will add a separate type for silent comments?

**Review by Goodwine — APPROVED** [review]

File: pkg/sass-parser/lib/src/statement/index.ts (hunk 3 above)
Comment (nex3) [reply pkg/sass-parser/lib/src/statement/index.ts:58]: Exactly.

**Pull request merged**